src/python/test/test_extract_expression.py

```diff
@@ -236,6 +268,146 @@ def common_input_args_test(self, input_argument_dict, expected_error_message):
 
         self.assertEqual(expected_error_message, str(cm.exception).strip())
 
+    #
+    # Positive output tests
+    #
+
+    def test_output_data_format(self):
+        expected_result = """feature_id,sample_id,expression,strand
+            ENST00000450305,sample_2,50,+
+            ENST00000456328,sample_2,90,+
+            ENST00000488147,sample_2,90,-""".replace(" ","")
+        output_path = os.path.join(
+            self.general_output_dir, "extract_assay_expression_data.csv"
+        )
+        # Generate the formatted output file
+        write_expression_output(
+            output_path,
+            ",",
+            False,
+            self.vizserver_data_mock_response["results"],
+        )
+        # Read the output file back in and compare to expected result
+        # Since the test should fail if the formatting is wrong, not just if the data is wrong, we
+        # can do a simple string comparison
+        with open(output_path, "r") as infile:
+            data = infile.read()
+        self.assertEqual(expected_result.strip(),data.strip())
+
+
+    def test_output_sql_format(self):
+        sql_mock_response = {
+            "sql": "SELECT `expression_1`.`feature_id` AS `feature_id`, `expression_1`.`sample_id` AS `sample_id`, `expression_1`.`value` AS `expression`, `expr_annotation_1`.`strand` AS `strand` FROM `database_gypg8qq06j8kzzp2yybfbzfk__enst_short_multiple_assays2`.`expression` AS `expression_1` LEFT OUTER JOIN `database_gypg8qq06j8kzzp2yybfbzfk__enst_short_multiple_assays2`.`expr_annotation` AS `expr_annotation_1` ON `expression_1`.`feature_id` = `expr_annotation_1`.`feature_id` WHERE `expression_1`.`value` >= 1 AND `expr_annotation_1`.`chr` = '1' AND (`expr_annotation_1`.`end` BETWEEN 7 AND 250000000 OR `expr_annotation_1`.`start` BETWEEN 7 AND 250000000 OR `expr_annotation_1`.`end` >= 250000000 AND `expr_annotation_1`.`start` <= 7)"
+        }
+        expected_result = "SELECT `expression_1`.`feature_id` AS `feature_id`, `expression_1`.`sample_id` AS `sample_id`, `expression_1`.`value` AS `expression`, `expr_annotation_1`.`strand` AS `strand` FROM `database_gypg8qq06j8kzzp2yybfbzfk__enst_short_multiple_assays2`.`expression` AS `expression_1` LEFT OUTER JOIN `database_gypg8qq06j8kzzp2yybfbzfk__enst_short_multiple_assays2`.`expr_annotation` AS `expr_annotation_1` ON `expression_1`.`feature_id` = `expr_annotation_1`.`feature_id` WHERE `expression_1`.`value` >= 1 AND `expr_annotation_1`.`chr` = '1' AND (`expr_annotation_1`.`end` BETWEEN 7 AND 250000000 OR `expr_annotation_1`.`start` BETWEEN 7 AND 250000000 OR `expr_annotation_1`.`end` >= 250000000 AND `expr_annotation_1`.`start` <= 7)"
+        output_path = os.path.join(
+            self.general_output_dir, "extract_assay_expression_sql.csv"
+        )
+        # Generate the formatted output file
+        write_expression_output(
```

Comment:
Missing 1 required positional argument: 'output_listdict_or_string' (`sql_mock_response` should be passed to `output_listdict_or_string`)